modules/aws_postgres/tf_module/rds.tf

```
@@ -21,18 +21,20 @@ resource "aws_rds_cluster" "db_cluster" {
   count                   = var.existing_global_database_id == null ? 1 : 0
   cluster_identifier      = "opta-${var.layer_name}-${var.module_name}-${random_string.db_name_hash.result}"
   db_subnet_group_name    = "opta-${var.env_name}"
-  database_name           = var.database_name
+  database_name           = var.restore_from_snapshot == null ? var.database_name : null
   engine                  = "aurora-postgresql"
   engine_version          = var.engine_version
-  master_username         = "postgres"
-  master_password         = random_password.pg_password.result
+  master_username         = var.restore_from_snapshot == null ? "postgres" : null
```

```suggestion
  master_username         = "postgres"
```

`master_username` is ignored when restoring from a snapshot, so keeping the "postgres" value is safe.

---

source?

---

[AWS provider source code](https://github.com/hashicorp/terraform-provider-aws/blob/8b5a71bcdcaa04e6d5c5c41243fd981edbdd79a5/internal/service/rds/cluster.go#L542). Basically, `master_username` is not read when restoring from snapshot, but it *is* marked as `ForceNew`, so if it changes (such as if the `restore_from_snapshot` variable is unset), it will force a recreation of the aurora cluster.